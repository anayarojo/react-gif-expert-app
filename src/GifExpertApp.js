import React, { useState } from 'react';
import PropTypes from 'prop-types';

import AddCategory from './components/AddCategory';
import GifCategoryGrid from './components/GifCategoryGrid';

const GifExpertApp = ({ defaultCategories }) => {

    // const [categories, setCategories] = useState(['One Punch']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h1>Gif Expert App</h1>
            <AddCategory setCategories={ setCategories } />
            <hr className="mb-5" />
            {
                categories.map((category) => (
                    <GifCategoryGrid 
                        key={ category } 
                        category={ category } 
                    />
                ))
            }
        </>
    );
}

GifExpertApp.propTypes = {
    defaultCategories: PropTypes.array,
};

GifExpertApp.defaultProps = {
    defaultCategories: ['One Punch'],
};

export default GifExpertApp;
