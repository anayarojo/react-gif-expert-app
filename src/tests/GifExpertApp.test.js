import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';

import GifExpertApp from '../GifExpertApp';

describe('Test for <GifExpertApp />', () => {
    
    test('Should show <GifExpertApp /> successfully', () => {

        const wrapper = shallow(<GifExpertApp />);
        
        expect(wrapper).toMatchSnapshot();
    });

    test('Should has <AddCategory /> component', () => {

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper.find('AddCategory').exists()).toBe(true);
    });

    test('Should has two items of <GifCategoryGrid /> component', () => {

        const categories = ['Naruto', 'One Piece']
        const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />);

        expect(wrapper.find('GifCategoryGrid').length).toBe(categories.length);
    });

});
