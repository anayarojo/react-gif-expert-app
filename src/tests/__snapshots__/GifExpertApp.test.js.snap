// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <GifExpertApp /> Should show <GifExpertApp /> successfully 1`] = `
<Fragment>
  <h1>
    Gif Expert App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr
    className="mb-5"
  />
  <GifCategoryGrid
    category="One Punch"
    key="One Punch"
  />
</Fragment>
`;
